rest/adminapitest/admin_api_test.go

```diff
@@ -2141,6 +2141,73 @@ func TestHandleCreateDB(t *testing.T) {
 	rest.RequireStatus(t, resp, http.StatusBadRequest)
 }
 
+func TestHandleCreateDBJsonName(t *testing.T) {
+	if base.UnitTestUrlIsWalrus() {
+		t.Skip("This test only works against Couchbase Server")
+	}
+	testCases := []struct {
+		name        string
+		JSONname    string
+		expectError bool
+	}{
+		{
+			name:        "Name match",
+			JSONname:    "db1",
+			expectError: false,
+		},
+		{
+			name:        "Name mismatch",
+			JSONname:    "dummy",
+			expectError: true,
+		},
+		{
+			name:        "No JSON Name",
+			JSONname:    "",
+			expectError: false,
+		},
+	}
+
+	for _, test := range testCases {
+		t.Run(test.name, func(t *testing.T) {
+			tb := base.GetTestBucket(t)
+			defer tb.Close()
+
+			rt := rest.NewRestTester(t, &rest.RestTesterConfig{
+				CustomTestBucket: tb,
+				PersistentConfig: true,
+				MutateStartupConfig: func(config *rest.StartupConfig) {
+					config.Bootstrap.ConfigUpdateFrequency = base.NewConfigDuration(0)
+				},
+			})
+			defer rt.Close()
+
+			var resp *rest.TestResponse
+			if test.name != "No JSON Name" {
+				resp = rt.SendAdminRequest(http.MethodPut, "/db1/",
+					fmt.Sprintf(
+						`{"bucket": "%s", "name": "%s", "num_index_replicas": 0, "enable_shared_bucket_access": %t, "use_views": %t}`,
+						tb.GetName(), test.JSONname, base.TestUseXattrs(), base.TestsDisableGSI(),
+					),
+				)
+			} else {
+				resp = rt.SendAdminRequest(http.MethodPut, "/db1/",
+					fmt.Sprintf(
+						`{"bucket": "%s", "num_index_replicas": 0, "enable_shared_bucket_access": %t, "use_views": %t}`,
+						tb.GetName(), base.TestUseXattrs(), base.TestsDisableGSI(),
+					),
+				)
+			}
+			if test.name == "Name mismatch" {
```

Comment:
It would be better to instead use the if statement to check `test.expectError` instead of using the test name to decide the assert.